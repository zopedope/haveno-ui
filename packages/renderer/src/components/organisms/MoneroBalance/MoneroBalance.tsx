// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { Group, createStyles } from "@mantine/core";
import { ReactComponent as MoneroIcon } from "@assets/monero.svg";
import { DetailItem } from "@atoms/DetailItem";

interface MoneroBalanceProps {
  children: React.ReactNode;
}
export function MoneroBalance({ children }: MoneroBalanceProps) {
  const { classes } = useStyles();

  return (
    <Group className={classes.root} spacing={0}>
      <MoneroIcon className={classes.moneroIcon} />
      <Group className={classes.content}>{children}</Group>
    </Group>
  );
}

interface MoneroBalanceDetail {
  label: string;
  children: React.ReactNode;
}

MoneroBalance.Detail = MoneroBalanceDetail;

function MoneroBalanceDetail({ ...props }: MoneroBalanceDetail) {
  return <DetailItem {...props} />;
}

const useStyles = createStyles((theme) => ({
  root: {
    backgroundColor: theme.white,
    border: `1px solid ${theme.colors.gray[3]}`,
    borderRadius: theme.radius.md,
    paddingTop: theme.spacing.md,
    paddingBottom: theme.spacing.md,
    paddingLeft: theme.spacing.md,
    paddingRight: theme.spacing.md,
  },
  moneroIcon: {
    height: 32,
    marginLeft: theme.spacing.xs,
    marginRight: theme.spacing.xl,
    width: 32,
  },
  content: {
    gap: theme.spacing.xl * 1.5,
  },
}));
