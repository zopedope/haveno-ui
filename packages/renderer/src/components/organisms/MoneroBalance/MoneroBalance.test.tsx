// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { MoneroBalance } from "./MoneroBalance";

describe("organisms::MoneroBalance", () => {
  it("renders without exploding.", () => {
    const { asFragment, unmount } = render(
      <MoneroBalance>
        <MoneroBalance.Detail label="Balance">10.20</MoneroBalance.Detail>
      </MoneroBalance>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("renders children content.", () => {
    const { unmount } = render(<MoneroBalance>Content here...</MoneroBalance>);

    expect(screen.queryByText("Content here...")).toBeInTheDocument();
    unmount();
  });

  it("renders monero balance detail", () => {
    const { unmount } = render(
      <MoneroBalance>
        <MoneroBalance.Detail label="Balance label">10.20</MoneroBalance.Detail>
      </MoneroBalance>
    );
    expect(screen.queryByText("Balance label")).toBeInTheDocument();
    expect(screen.queryByText("10.20")).toBeInTheDocument();
    unmount();
  });

  it("display Monero logo", () => {
    const { unmount, container } = render(
      <MoneroBalance>Content here...</MoneroBalance>
    );
    expect(container.querySelector("svg#monero-icon")).toBeInTheDocument();
    unmount();
  });
});
