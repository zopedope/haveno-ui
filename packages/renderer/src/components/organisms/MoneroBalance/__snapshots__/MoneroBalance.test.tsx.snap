// Vitest Snapshot v1

exports[`organisms::MoneroBalance > renders without exploding. 1`] = `
<DocumentFragment>
  <div
    class="mantine-Group-root mantine-3iq866"
  >
    <svg
      class="mantine-Group-child mantine-154defe"
      fill="none"
      height="1em"
      id="monero-icon"
      viewBox="0 0 32 32"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0_2924_33103)"
      >
        <path
          d="M15.9979 0C7.16414 0 -0.0112325 7.17316 1.32008e-05 15.9979C0.00222827 17.7635 0.283798 19.4618 0.812262 21.0505H5.5992V7.59219L15.9979 17.9897L26.3961 7.59219V21.0507H31.1841C31.7133 19.4622 31.9934 17.7638 31.9966 15.9982C32.0117 7.16422 24.8326 0.00212965 15.9979 0.00212965V0Z"
          fill="#111111"
        />
        <path
          d="M13.6084 20.3801L9.0703 15.8423V24.3109H5.60074L2.32678 24.3115C5.13506 28.9184 10.2103 32 15.9998 32C21.7893 32 26.8648 28.9177 29.6736 24.3108H22.9283V15.8423L18.39 20.3801L15.9993 22.7705L13.6086 20.3801H13.6084Z"
          fill="#111111"
        />
      </g>
      <defs>
        <clippath
          id="clip0_2924_33103"
        >
          <rect
            fill="white"
            height="32"
            width="32"
          />
        </clippath>
      </defs>
    </svg>
    <div
      class="mantine-Group-root mantine-Group-child mantine-1gdejmd"
    >
      <div
        class="mantine-Stack-root mantine-DetailItem-root mantine-Group-child mantine-2d5onn"
      >
        <div
          class="mantine-Text-root mantine-DetailItem-label mantine-8r3ub3"
        >
          Balance
        </div>
        <div
          class="mantine-Text-root mantine-DetailItem-content mantine-m3tadk"
        >
          10.20
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
